test/testdrive/createdrop.td

```diff
@@ -226,3 +226,19 @@ a
   FORMAT BYTES;
 
 > DROP SOURCE f
+
+# This tests Materialize's logic around custom AWS regions
+! CREATE SOURCE custom_source
+  FROM KINESIS ARN 'arn:aws:kinesis:custom-region::stream/fake-stream'
+  WITH (access_key = 'fake_access_key', secret_access_key = 'fake_secret_access_key')
+  FORMAT BYTES;
+If providing a custom region, an `endpoint` option must also be provided
+
+> CREATE SOURCE custom_source
+  FROM KINESIS ARN 'arn:aws:kinesis:custom-region::stream/fake-stream'
+  WITH (access_key = 'fake_access_key',
+        secret_access_key = 'fake_secret_access_key',
+        endpoint = 'fake_endpoint')
+  FORMAT BYTES;
+
+> DROP SOURCE custom_source
```

Comment:
Can you add this instead to a new file called “kinesis-sources”? I’m trying to keep the testdrive files more organized than they have been!

Reply:
Also no need to explicitly drop the source; testdrive will clean up as necessary. Leaving the source around is often nice for debugging, because you can run the testdrive script and then poke at the source in Materialize.